src/config/backends/power-fan.ts

```diff
@@ -0,0 +1,183 @@
+			'jetson-orin-nx-seeed-j4012',
+			'jetson-orin-nx-xavier-nx-devkit',
+		]).has(deviceType);
+	}
+
+	public async getBootConfig(): Promise<ConfigOptions> {
+		// Get raw config.json contents
+		let rawConf: unknown;
+		try {
+			rawConf = await this.configJson.get('os');
+		} catch (e: unknown) {
+			log.error(
+				`Failed to read config.json while getting power / fan configs: ${(e as Error).message ?? e}`,
+			);
+			return {};
+		}
+
+		// Decode to power fan schema from object type, filtering out unrelated values
+		const powerFanConfig = PowerFanConfig.SCHEMA.decode(rawConf);
+
+		if (isRight(powerFanConfig)) {
+			const conf = powerFanConfig.right;
+			return {
+				...(!isNullOrUndefined(conf.power?.mode) && {
+					power_mode: conf.power.mode,
+				}),
+				...(!isNullOrUndefined(conf.fan?.profile) && {
+					fan_profile: conf.fan.profile,
+				}),
+			};
+		} else {
+			return {};
+		}
+	}
+
+	public async setBootConfig(opts: ConfigOptions): Promise<void> {
+		// Read raw configs for "os" key from config.json
+		let rawConf;
+		try {
+			rawConf = await this.configJson.get('os');
+		} catch (err: unknown) {
+			log.error(`${(err as Error).message ?? err}`);
+			return;
+		}
+
+		// Decode to "os" object type while leaving in unrelated values
+		const maybeCurrentConf = schemaTypes.os.type.decode(rawConf);
+		if (!isRight(maybeCurrentConf)) {
+			log.error(
+				'Failed to decode current os config:',
+				Reporter.report(maybeCurrentConf),
+			);
+			return;
+		}
+		// Current config could be undefined if there's no os key in config.json, so default to empty object
+		const currentConf = maybeCurrentConf.right ?? {};
+
+		// Filter out unsupported options in target config
+		const supportedOpts = Object.fromEntries(
+			Object.entries(opts).filter(([key]) => this.isSupportedConfig(key)),
+		) as { power_mode?: string; fan_profile?: string };
+
+		const targetConf = structuredClone(currentConf);
+
+		// Update or delete power mode
+		if ('power_mode' in supportedOpts) {
+			targetConf.power = {
+				mode: supportedOpts.power_mode,
+			};
+		} else {
+			delete targetConf?.power;
+		}
+
+		// Update or delete fan profile
+		if ('fan_profile' in supportedOpts) {
+			targetConf.fan = {
+				profile: supportedOpts.fan_profile,
+			};
+		} else {
+			delete targetConf?.fan;
+		}
+
+		await this.configJson.set({ os: targetConf });
+	}
+
+	public isSupportedConfig = isSupportedConfig;
+
+	public isBootConfigVar(envVar: string): boolean {
+		return PowerFanConfig.CONFIGS.has(PowerFanConfig.stripPrefix(envVar));
+	}
+
+	public async isRebootRequired(opts: ConfigOptions): Promise<boolean> {
+		const supportedOpts = _.pickBy(
+			_.mapKeys(opts, (_value, key) => PowerFanConfig.stripPrefix(key)),
+			(_value, key) => this.isSupportedConfig(key),
+		);
+		const current = await this.getBootConfig();
+		// A reboot is only required if the power mode is changing
+		return current.power_mode !== supportedOpts.power_mode;
+	}
```

Comment:
Why was it decided to bother with this? Before, all host config changes would require a reboot. Even if it is not necessary for the fan profile, it doesn't seem worth it the extra complexity just to avoid a reboot in that case.

Reply:
Why does it cause storage wear? How often do we expect users to be changing the fan control speed? 

I agree that there are use cases for changing configurations without a reboot. 

I don't think the config variable interface is great for that. Config variables require an API call and a target state pull before the changes are applied, is not an interface made for runtime requirements.

If there is need for such an interface then we should work on that rather than trying to fit the config var interface for a specific use case

Reply:
Agreed, I think this feature has exposed some limits of the way we handle configs on top of what we were aware of before.